src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const res = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case add:
+        Object.assign(stateCopy, action.extraData);
+        res.push(Object.assign({}, stateCopy));
+        break;
+      case remove:
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        res.push(Object.assign({}, stateCopy));
+        break;
+      default:
+        for (const prop in stateCopy) {
+          delete stateCopy[prop];
+        }
+        res.push({});
+        break;
```

Comment:
This is not default case, it's "clear" action type. Default case should throw error informing user that provided action type is not supported if it's neither add, remove nor clear